eng/tools/generator/typespec/tspconfig.go

```diff
@@ -104,6 +104,14 @@ func ParseTypeSpecConfig(tspconfigPath string) (*TypeSpecConfig, error) {
 
 		goOption["module"] = module
 		tspConfig.EditOptions(string(TypeSpec_GO), goOption, false)
+
+		typespecGoOption, err := NewGoEmitterOptions(goOption)
```

Comment:
How about moving this 'module' validation to the beginning part at line#90?

Reply:
line 97-103, Replace service-dir and package-dir so that module becomes the actual value